pkg/networkservice/chains/client/client.go

```diff
@@ -22,66 +22,106 @@ import (
 
 	"google.golang.org/grpc"
 
+	"github.com/google/uuid"
 	"github.com/networkservicemesh/api/pkg/api/networkservice"
 
-	"github.com/networkservicemesh/sdk/pkg/networkservice/common/authorize"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/common/heal"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/common/mechanismtranslation"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/null"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/common/refresh"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/common/serialize"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/common/updatepath"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/core/chain"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/utils/metadata"
 )
 
+type clientOptions struct {
+	name                    string
+	onHeal                  *networkservice.NetworkServiceClient
+	additionalFunctionality []networkservice.NetworkServiceClient
+	authorizeClient         networkservice.NetworkServiceClient
+}
+
+// Option modifies default client chain values.
+type Option func(c *clientOptions)
+
+// WithHeal sets heal for the client.
+func WithHeal(onHeal *networkservice.NetworkServiceClient) Option {
+	return Option(func(c *clientOptions) {
+		c.onHeal = onHeal
+	})
+}
+
+// WithName sets name for the client.
+func WithName(name string) Option {
+	return Option(func(c *clientOptions) {
+		c.name = name
+	})
+}
+
+// WithAdditionalFunctionality sets additionalFunctionality for the client. Note: this adds into tail of the client chain.
+func WithAdditionalFunctionality(additionalFunctionality ...networkservice.NetworkServiceClient) Option {
+	return Option(func(c *clientOptions) {
+		c.additionalFunctionality = additionalFunctionality
+	})
+}
+
+// WithAuthorizeClient sets authorizeClient for the client chain.
+func WithAuthorizeClient(authorizeClient networkservice.NetworkServiceClient) Option {
+	if authorizeClient == nil {
+		panic("authorizeClient cannot be nil")
```

Comment:
Could we provide a bit more detail in the panic message about how to fix the problem (use WithAuthorizeClient etc?)

Reply:
@edwarnicke 

This message will be show when will be called:
```go
client.NewClient(ctx, conn, client.WithAuthorizeClient(nil))
```